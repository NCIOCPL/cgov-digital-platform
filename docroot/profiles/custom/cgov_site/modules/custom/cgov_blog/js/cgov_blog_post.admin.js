/**
 * @file
 * Placeholder file for custom blog post form behaviors.
 * TODO: make selectors settings 
 * 
 */
(function($, Drupal) {

  /**
   * @type {Drupal~behavior}
   * @prop {Drupal~behaviorAttach} attach
   *   Set behaviors for the admin Blog Post create / edit form.
   */
  Drupal.behaviors.blogPostForm = {
    attach: function (context, settings) {

      //console.log(settings.adminJS.selectable);
      //console.log(settings.adminJS.idPairs);

      /** Initialize correct selections on ready(). */
      $(function () {
        filterTopicOptions(settings);
      });

      /** Clear checks and show correct selections on dropdown change. */
      $( "#edit-field-blog-series" ).change(function() {
        $('#edit-field-blog-topics .form-checkbox').prop('checked', false);
        filterTopicOptions(settings);
      });

      /**
       * If the topic ID matches the series ID, remove the hidden class and show the checkboxes.
       * 
       * @param {*} settings 
       *    Drupalsettings
       */
      function filterTopicOptions(settings) {
        // Hide checkboxes to start.
        $('#edit-field-blog-topics .js-form-item').addClass('show-for-sr');

        // Get the seleccted series ID.
        var sid = $('select#edit-field-blog-series')[0].value;

        // Show every checkbox item that matches the owner series.
        var topics = settings.adminJS.idPairs;
        for (var tid in topics) {
          if(topics[tid] === sid) {
            jQuery('input[value=' + tid + ']').parent().removeClass('show-for-sr');
          }
        }
      } // filterTopicOptions()

    } // attach()
    
  }; // Drupal.behaviors.blogPostForm {}

})(jQuery, Drupal);