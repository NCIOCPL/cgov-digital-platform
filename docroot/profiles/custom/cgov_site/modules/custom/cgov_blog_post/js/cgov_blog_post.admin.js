(function($) {
  $.fn.drawBlogTopics = function(topics) {
    // Uncheck and hide all checkboxes when this function is called.
    $('#edit-field-blog-topics .js-form-item').addClass('visually-hidden');
    $('#edit-field-blog-topics .form-checkbox').prop('checked', false);

    // Get the seleccted series ID.
    var sid = $('select#edit-field-blog-series')[0].value;

    // If the topic ID matches the series ID, remove the hidden class and show the checkboxes.
    for (var tid in topics) {
      if(topics[tid] === sid) {
        jQuery('input[value=' + tid + ']').parent().removeClass('visually-hidden');
      }
    }

  };
  // TODO: hide topics wrapper if not in categories
  // TODO: clear checkmarks on switch & select other
  // TODO: make visually-hidden a wrapper
  // TODO: follow Drupal JS coding/naming standards
})(jQuery);
