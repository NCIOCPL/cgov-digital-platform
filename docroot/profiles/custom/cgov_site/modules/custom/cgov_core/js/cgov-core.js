/**
 *  @file
 */

 (function ($, Drupal) {
  Drupal.behaviors.cgovcore = {
    attach: function (context) {
    $('#edit-field-pretty-url-0-value').after(jQuery("<div id='url-data-value'>"));
    $('#edit-field-pretty-url-0-value').keyup(function () {
      if(this.value.length > 200){
        $("#url-data-value").html(this.value.substring(200) + ' - will be removed.');
      } else {
        $("#url-data-value").html('');
      }
    });
  }
  };
})(jQuery, Drupal);
