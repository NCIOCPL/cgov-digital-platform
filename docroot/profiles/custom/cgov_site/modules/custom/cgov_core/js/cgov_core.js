/**
 *  @file
 */

 (function ($, Drupal) {
  Drupal.behaviors.cgov_core = {
    attach: function (context) {
      $("#url-data-value").remove();
      $('#edit-field-pretty-url-0-value').after(jQuery("<div id='url-data-value' style='font-weight:bold'>"));
      showWarningMessage();
      $('#edit-field-pretty-url-0-value').keyup(function () {
        showWarningMessage();
      });

      // Helper function to show warning message.
      function showWarningMessage() {
        $this = $('#edit-field-pretty-url-0-value');
        if ($("div.item-container").attr("site-section-data-path")) {
          let total_length = $("div.item-container").attr("site-section-data-path").length + $this.val().length;
          // Check for exceeding the character count of 200 limit.
          // Combine site section and pretty url.
          if (total_length > 200) {
            let remaining_length = 200 - $("div.item-container").attr("site-section-data-path").length;
            $("#url-data-value").html($this.val().substring(remaining_length) + ' - will be removed from URL.');
          } else {
            $("#url-data-value").html('');
          }
        }
      }
    }
  };
})(jQuery, Drupal);
