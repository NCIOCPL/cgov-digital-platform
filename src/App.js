import React, { Fragment } from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import './styles/main.scss';

import SearchPage from './views/SearchPage';
import ResultsPage from './views/ResultsPage';

function App() {
  return (
    <Fragment>
      <Switch>
        <Redirect exact from="/" to="/search" />
        <Route path="/search" component={SearchPage} />
        <Route path="/r" component={ResultsPage} />
      </Switch>
    </Fragment>
  );
}

export default App;
