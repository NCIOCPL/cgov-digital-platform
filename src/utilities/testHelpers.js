import React from 'react';
import { MemoryRouter, Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';

export const TEST_ELEMENT_ID = 'test';

// renders an element in a test `document` and returns it.
export const getDomElement = element => {
  const instance = mount(<MemoryRouter>{element}</MemoryRouter>);
  const renderedElement = instance.getDOMNode();

  // Jest ships with `jsdom` which simulates a DOM environment. this environment
  // is cleared on every file run as each is its own thread... so, `document` will
  // reset per test file.
  document.getElementById(TEST_ELEMENT_ID).appendChild(renderedElement);

  return renderedElement;
};

// sets up a test DOM - a container to populate in tests.
export const setupDom = () => {
  document.body.innerHTML = `<div id="${TEST_ELEMENT_ID}"></div>`;
};

// restets a test DOM.
export const resetDom = () => {
  document.body.innerHTML = '';
};

export const renderWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  };
};
