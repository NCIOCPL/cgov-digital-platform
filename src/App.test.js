import React from 'react';
import { renderWithRouter } from './utilities/testHelpers.js';
import App from './App';
import { italic } from 'ansi-colors';
const defaultProps = {};


// mock page templates instead of using real components
jest.mock('./views/ResultsPage', () => () => (
  <div id="mockResultsPage">mockResultsPage</div>
));
jest.mock('./views/SearchPage', () => () => (
  <div id="mockSearchPage">mockSearchPage</div>
));

describe('App', () => {
  describe('Router', () => {
    it('routes to search page by default', () => {
      const {container, getByText} = renderWithRouter(<App />);
      expect(getByText('mockSearchPage')).toBeTruthy();
      expect(container).toMatchSnapshot();
    });

    it('routes to results when given route /r', () => {
      const {container, getByText} = renderWithRouter(<App />, {
        route: '/r'
      });
      expect(getByText('mockResultsPage')).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
});
