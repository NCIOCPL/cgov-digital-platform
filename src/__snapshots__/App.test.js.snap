// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Router routes to results when given route /r 1`] = `
<div>
  <div
    id="mockResultsPage"
  >
    mockResultsPage
  </div>
</div>
`;

exports[`App Router routes to search page by default 1`] = `
<div>
  <div
    id="mockSearchPage"
  >
    mockSearchPage
  </div>
</div>
`;
